File: 0000-0999/0150.EvaluateReversePolishNotation.go
func evalRPN(tokens []string) int {
    stack := make([]int, 0, len(tokens))
    for _, t := range tokens {
        num, ok := strconv.Atoi(t)
        if ok == nil {
            stack = append(stack, num)
        } else {
            n := len(stack)
            if t == "+" {
                stack[n-2] += stack[n-1]
            } else if t == "-" {
                stack[n-2] -= stack[n-1]
            } else if t == "*" {
                stack[n-2] *= stack[n-1]
            } else if t == "/" {
                stack[n-2] /= stack[n-1]
            }
            stack = stack[:n-1]
        }
    }
    return stack[0]
}

File: 0000-0999/0150.EvaluateReversePolishNotation.py
class Solution(object):
    def evalRPN(self, tokens):
        """
        :type tokens: List[str]
        :rtype: int
        """
        stack = []
        for n in tokens:
            if n not in '+-*/':
                stack.append(int(n))
            elif n in '+-*/' and len(stack) >= 2:
                b = stack.pop()
                a = stack.pop()
                if n == '+':
                    stack.append(a + b)
                elif n == '-':
                    stack.append(a - b)
                elif n == '*':
                    stack.append(a * b)
                elif b != 0:
                    stack.append(int(float(a) / b))
                    '''
                    if (a >= 0 and b > 0) or (a <= 0 and b < 0):
                        stack.append(a / b)
                    else:
                        stack.append(-(abs(a) / abs(b)))
                    '''
        return stack[0]

test = Solution()
print test.evalRPN(["10","6","9","3","+","-11","*","/","*","17","+","5","+"])
